src/lib/interlink.lua

```diff
@@ -130,15 +135,20 @@ local DOWN = 4 -- Both ends have detached; must be re-allocated.
 -- (any)    DXUP->DOWN  Cannot shutdown queue while it is in use.
 -- (any)    DOWN->*     Cannot transition from DOWN (must create new queue.)
 
-local function attach (name, initialize)
+local function attach (name, size, transitions)
+   assert(band(size, size-1) == 0, "size is not a power of two")
    local r
    local first_try = true
    waitfor(
       function ()
          -- Create/open the queue.
-         r = shm.create(name, "struct interlink")
-         -- Return if we succeed to initialize it.
-         if initialize(r) then return true end
+         r = shm.create(name, "struct interlink", size)
+         -- Initialize queue (only one process can set size).
+         if sync.cas(r.state, INIT, FREE) then
+            r.size = size
```

Comment:
Looking at this again I think this change introduces a race condition, because another process can transition `FREE->*` and assert the interlink size before it is set. Need an additional state?

Reply:
Should be fixed in [052d8c2](https://github.com/snabbco/snabb/pull/1489/commits/052d8c256d6b440031c3a05b902107bac0151d63)